IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -147,12 +169,20 @@ For asymmetric keys this holds the public key hash which allows digital identity
     }
     
     NSMutableDictionary *mutableCertQuery = [NSMutableDictionary new];
-    
     if (certAttributes)
     {
         [mutableCertQuery addEntriesFromDictionary:certAttributes];
     }
     
+#if TARGET_OS_OSX
+    // For macOS, if the key is ECC key, use shared access group to query certificate. If it is RSA, remove shared access group from query for certificate as it is not login keychain.
+        NSString *sharedAccessGroup = [certAttributes valueForKey:(__bridge id)kSecAttrAccessGroup];
+        if (sharedAccessGroup && ![[MSIDKeyOperationUtil sharedInstance] isKeyFromSecureEnclave:privateKeyRef])
+        {
+            [mutableCertQuery removeObjectForKey:(__bridge id)kSecAttrAccessGroup];
+        }
+#endif
+    
     mutableCertQuery[(__bridge id)kSecClass] = (__bridge id)kSecClassCertificate;
```

Comment:
For the sake of clean code, I think you can break it into 2 methods: one to query private key, then one for querying cert.

Reply:
Re-thinking this, since the purpose of the method is to return Certificate + Private Key as MSIDWPJKeyPairWithCert we are always going to call code to get certificate here. I don't see any other class/internal method calling a function to get certificate so splitting it into another method won't be beneficial.